Source/SwiftLintFramework/Extensions/File+Cache.swift

```diff
@@ -52,20 +67,68 @@ private struct Cache<T> {
 
 extension File {
 
+    private var cacheKey: String {
+        return path ?? "\(ObjectIdentifier(self).hashValue)"
+    }
+
+    internal var sourcekitdFailed: Bool {
+        get {
+            return responseCache.get(self) == nil
+        }
+        set {
+            if newValue {
+                responseCache.values[cacheKey] = Optional<[String: SourceKitRepresentable]>.None
+            } else {
+                responseCache.values.removeValueForKey(cacheKey)
+            }
+        }
+    }
+
+    internal var assertHandler: (() -> ())? {
+        get {
+            return assertHandlers[cacheKey] ?? nil
+        }
+        set {
+            assertHandlers[cacheKey] = newValue
+        }
+    }
+
     internal var structure: Structure {
-        return structureCache.get(self)
+        guard let structure = structureCache.get(self) else {
+            if let handler = assertHandler {
+                handler()
+                return Structure(sourceKitResponse: [:])
+            }
+            fatalError("Never call this for file that sourcekitd fails.")
+        }
+        return structure
     }
 
     internal var syntaxMap: SyntaxMap {
-        return syntaxMapCache.get(self)
+        guard let syntaxMap = syntaxMapCache.get(self) else {
+            if let handler = assertHandler {
+                handler()
+                return SyntaxMap(data: [])
+            }
+            fatalError("Never call this for file that sourcekitd fails.")
```

Comment:
@norio-nomura what does this error message mean? I have hit this when running locally.

Reply:
It's an internal SwiftLint error for this to be invoked when SourceKit is in a state where it has failed. If you can provide a stack trace the next time you see this, we can defensively avoid it by adding a check to see if SourceKit has failed.